addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,371 @@
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link smhiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Parment - Initial contribution
+ */
+@NonNullByDefault
+public class SmhiHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SmhiHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    protected static final String smhiURL = "http://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/%s/lat/%s/data.json";
+
+    protected @Nullable PointType location;
+
+    // Timeout for weather data requests.
+    private static final int SMHI_TIMEOUT = 5000;
+
+    private Gson gson = new GsonBuilder().create();
+
+    private String apiRequest = "";
+    private String groupId = "";
+
+    @Nullable
+    private SmhiConfiguration config;
+
+    public SmhiHandler(Thing thing) {
+        super(thing);
+    }
+
+    private SmhiParameterTables parameterTable = new SmhiParameterTables();
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            queryApiAndUpdateChannels();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing SMHI weather handler.");
+
+        config = getConfigAs(SmhiConfiguration.class);
+
+        if (StringUtils.trimToNull(config.location) == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The 'location' parameter must be configured.");
+            return;
+        }
+
+        try {
+            location = new PointType(config.location);
+        } catch (IllegalArgumentException e) {
+            location = null;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The 'location' parameter could not be split into latitude and longitude.");
+            return;
+        }
+
+        DecimalFormat df = new DecimalFormat("##.######");
+        DecimalFormatSymbols custom = new DecimalFormatSymbols();
+        custom.setDecimalSeparator('.');
+        df.setDecimalFormatSymbols(custom);
+
+        String[] parts = config.location.split(",");
+        double dlatitude = Double.valueOf(parts[0]);
+        double dlongitude = Double.valueOf(parts[1]);
+        String latitude = df.format(Double.valueOf(parts[0]));
+        String longitude = df.format(Double.valueOf(parts[1]));
+
+        apiRequest = String.format(smhiURL, longitude, latitude);
+
+        try {
+            if (HttpUtil.executeUrl("GET", apiRequest, null, null, "application/json", SMHI_TIMEOUT)
+                    .contains("out of bounds")) {
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "The 'location' parameter is out of bound.");
+                return;
+
+            }
+

```

Comment:
Remove all trailing empty lines (also in other files). The SAT report should also show this.